src/main/java/org/cactoos/iterator/Endless.java

```diff
@@ -56,15 +56,15 @@ public Endless(final T element) {
      * Ctor.
      * @param scalar Scalar to repeat
      */
-    public Endless(final Scalar<T> scalar) {
+    public Endless(final Scalar<? extends T> scalar) {
         this(new Unchecked<>(scalar));
     }
 
     /**
      * Ctor.
      * @param scalar Scalar to repeat
      */
-    public Endless(final Unchecked<T> scalar) {
+    public Endless(final Unchecked<? extends T> scalar) {
```

Comment:
@DmitryBarskov This ctor should be either private or removed, there should be no concrete parameters in ctors